auth/src/main/java/org/aerogear/auth/RealmRole.java

```diff
@@ -1,7 +1,31 @@
 package org.aerogear.auth;
 
+/**
+ * Creates a new keycloak realm role for a user
+ */
 public final class RealmRole extends AbstractRole {
     public RealmRole(final String roleName) {
         super(roleName);
     }
-}
\ No newline at end of file
+
+    /**
+     * Get's the role type
+     *
+     * @return the role type
+     * @see RoleType
+     */
+    @Override
+    public RoleType getRoleType() {
+        return RoleType.REALM;
+    }
+
+    /**
+     * Get's the client id for the realm role
+     *
+     * @return null (realms don't have client ids)
+     */
+    @Override
+    public String getClientID() {
```

Comment:
If the RealmRole do not have a client ID than this method should not be here. Moreover, default implementations should be put in the AbstractRole class

Reply:
See comment above